openproblems/api/run.py

```diff
@@ -1,16 +1,27 @@
+from ..utils import temporary
 from . import utils
 
 import anndata
 
 
+@temporary(version="1.0")
+def _backport_code_version(adata, fun):
+    if "method_code_version" not in adata.uns:
+        adata.uns["method_code_version"] = fun.metadata["code_version"]
+    return adata
```

Comment:
ah, you thought of that already, nice :D.

Reply:
Why only do this for a particular version? Why not as standard always (and then not ask users to do this manually in the method functions?

Reply:
So the issue is that many packages are not installed in the base image, which is where the decorators are evaluated, which leads to a whole lot of ModuleNotFound errors. So instead we have to do the method check inside the function, which runs in the right docker image.

Reply:
I see. And if you define this in the decorator for a method function, that would not also be run in the docker image?

Just trying to think how to reduce work on method contributors and reduce review complexity.

Reply:
Yep, that's right -- the decorator runs outside the docker image, because one of the things the decorator does is select the docker image.

Reply:
ah, of course. Okay, thanks! This makes it more clear to me now.